### PR TITLE
DCOS-47182: checkout right branch before running tests

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
## Testing

- See in the jenkins log that the branch of this PR is checked out:

**Before**

This is the part on https://jenkins.mesosphere.com/service/jenkins/blue/organizations/jenkins/frontend%2Fdcos-ui-oss-pipeline/detail/PR-3509/7/pipeline/46

```
[cos-ui-oss-pipeline_PR-3509-L5OARLBWBH65RUTYTLUT6BX7NYIOMZI3YZ22NNM7Q52R62N4MONA] Running shell script

++ '[' -z master ']'

++ echo master

+ git checkout master

Already on 'master'

Your branch is up-to-date with 'origin/master'.
```

**After**

This is the part on https://jenkins.mesosphere.com/service/jenkins/blue/organizations/jenkins/frontend%2Fdcos-ui-oss-pipeline/detail/PR-3531/1/pipeline

```
[cos-ui-oss-pipeline_PR-3531-4RGCRZ6VLVYNQFTUHQTRTU2SUOJZ643PFPWSNWHBI4RFUZI5RGXA] Running shell script

++ '[' -z danielmschmidt/DCOS-47182 ']'

++ echo danielmschmidt/DCOS-47182

+ git checkout danielmschmidt/DCOS-47182

Switched to a new branch 'danielmschmidt/DCOS-47182'

Branch danielmschmidt/DCOS-47182 set up to track remote branch danielmschmidt/DCOS-47182 from origin.
```

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -38,7 +38,7 @@ pipeline {
           sh "rm -rf {.*,*} || ls -la && git clone https://\$GIT_USER:\$GIT_PASSWORD@github.com/dcos/dcos-ui.git ."
         }
         sh "git fetch"
-        sh "git checkout \"\$([ -z \"\$CHANGE_TARGET\" ] && echo \$BRANCH_NAME || echo \$CHANGE_TARGET )\""
+        sh "git checkout \"\$([ -z \"\$CHANGE_BRANCH\" ] && echo \$BRANCH_NAME || echo \$CHANGE_BRANCH )\""
```
Comment (juliangieseke) [inline Jenkinsfile]: `git merge` is missing 🤔

Comment (DanielMSchmidt) [reply Jenkinsfile]: That was missing before, can we make it a not-emergency task to add it? We would have been living without it a month anyways. I would feel better if I knew we had the time to do this properly.

Comment (juliangieseke) [reply Jenkinsfile]: please create a follow-up PR. I'd say not having merge is worse than not linting commits…

Comment (juliangieseke) [reply Jenkinsfile]: actually… We're enforcing that PRs are up to date which ends up in us not having them to be merged with master 🤔

**Pull request merged**